crates/common/mqtt_channel/src/config.rs

```diff
@@ -61,8 +64,37 @@ pub struct Config {
     /// Default: None
     pub initial_message: Option<InitMessageFn>,
 
-    /// TLS configuration used to connect to the broker.
-    pub cert_store: Option<rustls::RootCertStore>,
+    /// Certificate authentication configuration
+    pub authentication_config: Option<AuthenticationConfig>,
+}
+
+/// MQTT certificate authentication configuration.
+///
+/// Intended to mirror authentication model found in the mosquitto MQTT broker,
+/// when certificate-based authentication is enabled, server authentication is
+/// mandatory and client authentication is optional.
```

Comment:
The comment is not clear here, does this mean when the `server` authentication is enabled `clients` can connect without authentication?

Reply:
there are 3 supported modes of connecting:

1. no authentication
2. server authentication - clients will verify MQTT broker certificate
3. server and client authentication - clients will verify MQTT broker certificate and broker will verify client certificates

Will reword the comment to make it more clear and include link to the relevant mosquitto.conf manual section